File: simulator/src/main/kotlin/com/demo/flight/simulator/models/FlightEvent.kt
package com.demo.flight.simulator.models

import java.time.Instant

/**
 * Represents a flight event in the system.
 */
data class FlightEvent(
    val flightId: String,
    val airline: String,
    val eventType: EventType,
    val timestamp: Instant,
    val latitude: Double,
    val longitude: Double,
    val delayMinutes: Int,
    val origin: String,
    val destination: String
)

/**
 * Types of flight events that can be generated.
 */
enum class EventType {
    POSITION_UPDATE,
    DELAY_NOTIFICATION,
    DEPARTURE,
    ARRIVAL,
    CANCELLATION
}

File: frontend/src/models/FlightEvent.ts
/**
 * Interface representing a flight event from the backend API
 */
export interface FlightEvent {
  flightId: string;
  airline: string;
  eventType: 'POSITION_UPDATE' | 'TAKEOFF' | 'LANDING' | 'DELAY_UPDATE';
  timestamp: string;
  latitude: number;
  longitude: number;
  delayMinutes: number;
  origin: string;
  destination: string;
}

/**
 * Interface for alert items displayed in the alert feed
 */
export interface AlertItem {
  title: string;
  content: string;
  type: 'info' | 'success' | 'warning' | 'danger';
  timestamp: Date;
}
